pkg/resource/deploy/deploytest/provider.go

```diff
@@ -203,7 +205,11 @@ func (prov *Provider) Construct(info plugin.ConstructInfo, typ tokens.Type, name
 	if prov.ConstructF == nil {
 		return plugin.ConstructResult{}, nil
 	}
-	monitor, err := dialMonitor(context.Background(), info.MonitorAddress)
+	dialMonitorImpl := dialMonitor
+	if prov.DialMonitorF != nil {
```

Comment:
I wonder if this should actually just push this logic down into CallF/ConstructF. If we passed the address string to ConstructF/CallF and exposed DialMonitor for tests to use where that's appropriate, and for these new tests they just short circuit to only looking at the string.

Reply:
That's interesting. I'm not sure of other tests that expect `dialMonitor` will be called, but we should not change existing behavior by not calling it as this PR is focused on adding test coverage without modifying behavior.

Reply:
deploytest is literally just for our tests, it's fine to refactor

Reply:
> I wonder if this should actually just push this logic down into CallF/ConstructF. If we passed the address string to ConstructF/CallF and exposed DialMonitor for tests to use where that's appropriate, and for these new tests they just short circuit to only looking at the string.

Seems like a reasonable suggestion! However, I'm not sure if we can do this refactor and still hit our coverage goal by EOW. I suggest we go with `DialMonitorF` for now to continue to make progress, and follow-up with potential refactor after.